test/box/gh-5645-several-iproto-threads.test.lua

```diff
@@ -0,0 +1,50 @@
+env = require('test_run')
+net_box = require('net.box')
+fiber = require('fiber')
+test_run = env.new()
+test_run:cmd("create server test with script='box/gh-5645-several-iproto-threads.lua'")
+
+test_run:cmd("setopt delimiter ';'")
+function ddos(server_addr, fibers_count)
+    local fibers = {}
+    for i = 1, fibers_count do
+        fibers[i] = fiber.new(function()
+            local conn = net_box.new(server_addr)
+            for _ = 1, 100 do
+                pcall(conn.call, conn, 'ping')
+            end
+        end)
+        fibers[i]:set_joinable(true)
+    end
+    for _, f in ipairs(fibers) do
+        f:join()
+    end
+end;
+test_run:cmd("setopt delimiter ''");
+
+time_max = 20
+time_before = fiber.clock()
+fibers_count = 256
+test_run:cmd('start server test with args="1"')
+server_addr = test_run:cmd("eval test 'return box.cfg.listen'")[1]
+test_run:cmd("setopt delimiter ';'")
+while fiber.clock() - time_before < time_max do
+    ddos(server_addr, fibers_count)
+    if test_run:grep_log("test", "stopping input on connection") ~= nil then
+        break
+    end
+    fibers_count = fibers_count * 4
+end;
+test_run:cmd("setopt delimiter ''");
+test_run:cmd("stop server test")
+
+test_run:cmd('start server test with args="8"')
+server_addr = test_run:cmd("eval test 'return box.cfg.listen'")[1]
+fibers_count = fibers_count * 2
+assert(test_run:grep_log("test", "stopping input on connection") == nil)
```

Comment:
Could you please explain what is the purpose of the test and what are you checking?

Reply:
The reason why we are making this patch is that: there are users that have specific workloads where iproto thread
is the bottleneck of throughput: iproto thread's code is 100% loaded while TX thread's core is not. I don't know how to check full iproto thread loading in test, except for indirect signs, one of which is the message in the log "stopping connection on input". So in test first of all i start instance with one iproto thread, and send requests in a large number of fibers at the same time to this instance. I increase the number of fibers by 4 times until I get "stopping connection on input" in log. Then i restart instance with 8 itproto threads increase the number of fibers by 4 times again and check that there is no such message. Based on this, I conclude that 8 threads can withstand at least 4 times the more load

Reply:
If you have any ideas on how to test this better, that would be great!